packages/aws-cdk/lib/toolkit/error.ts

```diff
@@ -0,0 +1,48 @@
+const TOOLKIT_ERROR_SYMBOL = Symbol.for('@aws-cdk/core.TooklitError');
+const AUTHENTICATION_ERROR_SYMBOL = Symbol.for('@aws-cdk/core.AuthenticationError');
+
+/**
+ * Represents a general toolkit error in the AWS CDK Toolkit.
+ */
+class ToolkitError extends Error {
+  /**
+   * Determines if a given error is an instance of ToolkitError.
+   */
+  public static isToolkitError(x: any): x is ToolkitError {
+    return x !== null && typeof(x) === 'object' && TOOLKIT_ERROR_SYMBOL in x;
+  }
+
+  /**
+   * Determines if a given error is an instance of AuthenticationError.
+   */
+  public static isAuthenticationError(x: any): x is AuthenticationError {
+    return this.isToolkitError(x) && AUTHENTICATION_ERROR_SYMBOL in x;
+  }
+
+  /**
+   * The type of the error, defaults to "toolkit".
+   */
+  public readonly type: string;
```

Comment:
is this being used anywhere other than tests?

Reply:
Nope (not yet at least) - I am also copying the implementation from `ConstructError` again here: https://github.com/aws/aws-cdk/blob/8718767c5bf9fce33ea80e6c7d3bb136c79dbf7d/packages/aws-cdk-lib/core/lib/errors.ts#L56

Adding a `type` property is also the third instruction on Momo's ticket: https://github.com/aws/aws-cdk/issues/32347

Reply:
again i wish you had a real reason beyond Momo telling you to. but this is fine, i suppose